### DISCUSSION TITLE
Authenticating with keycloak constantly re-logging in over the course of a day

### DISCUSSION THREAD

**Discussion by brendanheywood** [outline/outline#7746] [General]
We have Keycloak SSO setup which has otherwise been quite functional. 

We are seeing lots of re-authentications over the course of a given day, which cause the pages to bounce around a bunch of redirects and because you are still logged in then it still works. It is weird because you may be actively using outline in another tab, then click on a link to open another doc and you will do the redirect bounce around. We typically aim to tune session inactivity timeouts to cover a full business day without re-prompting for login.

I have a suspicion that the culprit is the lastSignedIn cookie with value of oidc has SameSite=Strict which means when you click on a fresh link from another domain that cookie isn't visible to the server and it starts to reauthenticate. I'm not across the code but that doesn't fully explain it as I'd guess this logic only triggers if the main session is expired.

So what timeouts and config items can we tune to eliminate this re-login delay?

**Comment by jay-oswald** [comment]
I've also had some weirdness to do with SSO and multiple tabs and clicking on links.

For me I don't get asked to re sign in, already signed into Keycloak so the SSO redirect happens in the background, so it slows down the page load while all the SSO redirects happen.

However the main issue is sometimes (seams to be at certain times, not a chance, e.g. if I'm having the problem I will have it for every link, but other times no links are a problem) when clicking a link to open a new tab of outline, if I already have a background tab open, the new tab will get stuck in a SSO redirect loop, going from outline to keycloak in a loop, the only way I've found to fix it is either find my background tab and close it, then the SSO redirect loop fixes itself. Or copying the link and going directly to the outline URL, without clicking on the link also seams to work and load the new tab
